pkg/client/proxy_metrics.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"github.com/pkg/errors"
+
+	rpc "github.com/longhorn/longhorn-instance-manager/pkg/imrpc"
+)
+
+func (c *ProxyClient) MetricsGet(serviceAddress string) (metrics *Metrics, err error) {
+	input := map[string]string{
+		"serviceAddress": serviceAddress,
+	}
+	if err := validateProxyMethodParameters(input); err != nil {
+		return nil, errors.Wrap(err, "failed to get metrics for volume")
+	}
+
+	defer func() {
+		err = errors.Wrapf(err, "%v failed to get metrics for volume", c.getProxyErrorPrefix(serviceAddress))
+	}()
+
+	req := &rpc.ProxyEngineRequest{
+		Address: serviceAddress,
+	}
+	resp, err := c.service.MetricsGet(getContextWithGRPCTimeout(c.ctx), req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Metrics{
```

Comment:
It seems like we have similar types defined in different components like manager, instance-manager, engine or even replica. It's better to consolidate them in one place via protobuf, then reused together. A data model repo.  

Let's create a ticket for this refactoring.